### PR TITLE
display error message if app fails to initialize

### PR DESCRIPTION
**Pull request by UnsungHero97**

### CODE REVIEW

**Comment by solomontang** [comment]
This app initialization procedure seems to be a new pattern we want all apps to adopt. This PR also seems like these changes should also be added to `lets-react`

**Comment by UnsungHero97** [comment]
@solomontang, I totally agree, and it's already in `lets-react`.

File: src/core/edm/EDMReducer.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  getEntitySetIds,
+} from './EDMActions';
+
+// TODO: use models
+const { FullyQualifiedName } = Models;
+
+const INITIAL_STATE :Map<*, *> = fromJS({
+  entitySetIdsByName: Map(),
+  isFetchingAllEntitySetIds: false,
+  isFetchingAllTypes: false,
```
Comment (solomontang) [inline src/core/edm/EDMReducer.js]: Should the `isFetchingAll...` properties be `RequestStates` instead of `booleans`?

Comment (UnsungHero97) [reply src/core/edm/EDMReducer.js]: Yes. The only thing I haven't finished thinking through is how / when to reset the state back to `STANDBY`, but I know that it should definitely not be in the `FINALLY` block. Thoughts?

Comment (solomontang) [reply src/core/edm/EDMReducer.js]: Perhaps in the mount phase of the consuming component?

**Pull request merged**